src/Symfony/Bridge/Doctrine/Validator/Constraints/UniqueEntityValidator.php

```diff
@@ -62,6 +62,13 @@ public function validate($entity, Constraint $constraint)
             throw new ConstraintDefinitionException('At least one field has to be specified.');
         }
 
+        $target = $constraint->target;
+        /* @var $target string */
+        if (!$constraint->target) {
+            @trigger_error('Not providing a target for a UniqueEntity constraint is deprecated since version 3.1 and will be removed in 3.0.', E_USER_DEPRECATED);
```

Comment:
```diff
- @trigger_error('Not providing a target for a UniqueEntity constraint is deprecated since version 3.1 and will be removed in 3.0.', E_USER_DEPRECATED);
+ @trigger_error('Not providing a target for a UniqueEntity constraint is deprecated since version 3.1 and will be removed in 4.0.', E_USER_DEPRECATED);
```

(4.0, not 3.0)